src/main/kotlin/cacophony/semantic/analysis/ClosureAnalysis.kt

```diff
@@ -8,6 +8,9 @@ import cacophony.semantic.syntaxtree.LambdaExpression
  */
 typealias ClosureAnalysisResult = Map<LambdaExpression, Set<Variable>>
 
-fun analyseClosures(): ClosureAnalysisResult {
-    return emptyMap() // TODO - implement closure analysis
+fun analyseClosures(escapeAnalysis: EscapeAnalysisResult /*, lambdaAnalysis: LambdaAnalysisResult */): ClosureAnalysisResult {
+//    return lambdaAnalysis.mapValues { (_, analysis) ->
```

Comment:
Like above, if this actually works under some sensible interface of `LambdaAnalysisResult` then I think we can have this uncommented.